src/rust/catnip/runtime/memory/mempool.rs

```diff
@@ -50,7 +50,10 @@ impl MemoryPool {
 
         // Failed to create memory pool.
         if pool.is_null() {
-            return Err(Fail::new(libc::EAGAIN, "failed to create memory pool"));
+            let rte_errno: libc::c_int = unsafe { dpdk_rs::rte_errno() };
+            let cause: String = format!("failed to create memory pool: {:?}", rte_errno);
+            warn!("{}", cause);
```

Comment:
I would also include the name of the function in the warning messages, because if the same module has same error messages, we cannot distinguish between them.